homeassistant/components/systemmonitor/sensor.py

```diff
@@ -166,6 +183,20 @@ def update(self):
             self._state = round(psutil.swap_memory().free / 1024 ** 2, 1)
         elif self.type == "processor_use":
             self._state = round(psutil.cpu_percent(interval=None))
+        elif self.type == "processor_temperature":
+            temps = psutil.sensors_temperatures()
+            # There might be additional keys to be added for different
+            # platforms / hardware combinations.
+            # Raspberry 4
+            if "cpu-thermal" in temps:
+                self._state = round(temps["cpu-thermal"][0].current, 1)
+            # Linux default
+            elif "coretemp" in temps:
+                self._state = round(temps["coretemp"][0].current, 1)
+            # Fallback if no CPU sensor data available
+            else:
+                self._state = "N/A"
```

Comment:
I opted for `N/A` to show that this is not a setup issue (i.e. wrong config that simply cannot work), but to clearly show that the sensor was setup correctly and tried to retrieve temperature values but none are available in the hardware setup. Just showing an empty value (which would lead to a "Unknown" in the frontend) does not carry the same meaning.
@fabaff With that in mind, do you still prefer `None`?

Reply:
I commented it below, but `None` is correct. Availability is implemented using the `available` property of an entity.

Reply:
The usage of the `available` property in this context is wrong. The sensor can't recover during run-time. It will never become available. 

You have to check during the platform setup if the CPU is able to report the temperature or not and if not make the setup of the sensor fails and log it. The `glances` sensor had a list of the [most common CPU labels](https://github.com/home-assistant/core/commit/09f9875ccf8c7c48208c7ccaf148a90000d2e51f#diff-0b2cfe0523408f5a3c9b5790e0db5c4fR135). Use this as a start for the checks.

Reply:
I have already locally adjusted the `available` handling, but regarding the sensor data: 

Is it really the best idea to use the label to find the sensor data? I went for the temperature array key instead so far since I assumed that this would be more stable compared to a label meant for user display. You can even (at least for lm-sensor which I think is used as the data basis) overwrite the labels, so your machine might actually return different values.

Reply:
There is no need to use `available` in this case. Remove it. Check in `setup_platform` if the CPU is supported. If not, don't add the sensor for the CPU temperature.

The [list](https://github.com/shutupflanders/home-assistant/blob/44a6873427c930dc9b43cb1779d2906cccc1085e/homeassistant/components/glances/sensor.py#L199) of the `glances` was collected over the years and the intergrations (`systemmonitor` and `glances`) are using the same Python module to get the data. Thus, the chance is high that the same identifier for the CPU can be used.

Reply:
I personally quite like the entity to be added and visible so that if a configuration is shared across different machines it is visible even if the sensoring does not work.
![image](https://user-images.githubusercontent.com/114137/84525091-5f01f300-acdb-11ea-937b-aff904b9935d.png)

But I won't argue about that point and changed it. I now also re-used the old "glances" CPU temperature labels.

Reply:
@fabaff I checked again. The CPU part does not work on a Raspberry PI 4, because the "label" of `psutil` is empty. For `glances` it works, because they are apparently adding a label on their own (my RPI4 shows "cpu-thermal 1" as the label).

![image](https://user-images.githubusercontent.com/114137/84535025-43064d80-aceb-11ea-8a33-42b2d629bb8b.png)

My approach of testing for the sensor type/source instead worked fine on the RPI4.

Reply:
> I personally quite like the entity to be added and visible so that if a configuration is shared across different machines it is visible even if the sensoring does not work.

This will be no longer an issue if the integration is updated according [ADR003](https://github.com/home-assistant/architecture/blob/master/adr/0003-monitor-condition-and-data-selectors.md). Will require a couple of decisions (e.g., how to handle `disk` monitoring). Till then I agree that sharing the configuration across different CPU platforms could be a nice feature but as it's a one-time change we should start to support it

> The CPU part does not work on a Raspberry PI 4, because the "label" of psutil is empty. 

Not sure if that's based on the cause that the Raspberry Pi 4 is not supported by the mainline kernel yet.

Reply:
@fabaff Then to me it's not really clear how to proceed from here. Is ADR003 now blocking this PR, because the change is not exposing all temperature sensor data?

I am also not completely sure, how ADR003 would look like here. Would the integration need to provide all temperature sensor data to HA and then have it create an entity for each one (not only CPU)? That way HA would have to keep track of potentially many unwanted entities (performance impact?).

Reg. RPI4: Where does this leave us? I currently only use HA on RPI4 and virtualized machines (that therefore do never expose sensor data). If we stick with the copy from `glances` for the sensor labels, I cannot verify that it works and anyway have to wait with using "my2 new feature until the kernel changes (if that is actually the issue).

Reply:
@fabaff Can you please advise how to proceed with this PR?

Reply:
Reg. RPI4: I had a look at the `glances` coding (the actual tool, not the HA integration) and saw that they are constructing their own labels, if none is provided by the sensor (as is the case for RPI4). I now handle it the same for the `systemmonitor` HA integration.